packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -176,10 +176,25 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
       },
       {
         path: ['appointmentDateTime.recurringPatternEndDate'],
-        message: 'A recurring appointment should have an end date',
+        message: t('recurringAppointmentShouldHaveEndDate', 'A recurring appointment should have an end date'),
       },
-    );
+    )
+    .refine(
+      (formValues) => {
+        const appointmentDate = formValues.appointmentDateTime?.startDate;
+        const dateAppointmentScheduled = formValues.dateAppointmentScheduled;
 
+        if (!appointmentDate || !dateAppointmentScheduled) return true;
+        return dateAppointmentScheduled < appointmentDate;
```

Comment:
I'm wondering if the issue with the E2E tests here when the date scheduled is the same day as the appointment date, but earlier in time?

For instance, perhaps you only schedule appointments by the day (not day and time).  So say a 10am on August 16, you try to schedule an appointment for "August 16th", and because there is no time component it says it is *before* the scheduled date.  Could we just compare the "date" (as opposed to "datetime" components of the date here)?